apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -954,6 +957,13 @@ type ResourcePlacementStatus struct {
 	// +kubebuilder:validation:MaxItems=100
 	DiffedPlacements []DiffedResourcePlacement `json:"diffedPlacements,omitempty"`
 
+	// ObservedResourceIndex is the index of the resource snapshot that is currently being rolled out to the given cluster.
+	// During rollout, depending on the rollout strategy, clusters may observe different resource indices.
+	// ObservedResourceIndex is the resource snapshot index observed by the conditions in the ResourcePlacementStatus.
+	// This field is only meaningful if the `ClusterName` is not empty.
+	// +kubebuilder:validation:Optional
+	ObservedResourceIndex string `json:"observedResourceIndex,omitempty"`
```

Comment:
Another possible solution is that, we do not add this `ObservedResourceIndex` in each member cluster, while we only change the `ObservedResourceIndex` to the latest resourceSnapshot index version installed on the member clusters. This does not break current API.